ForwardingDecision.cs

```diff
@@ -35,14 +35,7 @@ public abstract class ForwardingDecision {
     /// </summary>
     public sealed class Drop : ForwardingDecision {
       // We only need to keep a single instance of Drop, so we use a singleton pattern.
-      public static readonly Drop inst;
-      private Drop () {}
```

Comment:
It's just that the default constructor is public, which is fine - but you made the constructor private as part of the singleton pattern, so I assume you removed it by accident?